core/src/main/java/com/scalar/db/transaction/consensuscommit/CrudHandler.java

```diff
@@ -97,11 +108,11 @@ private void read(Snapshot.Key key, Get get) throws CrudException {
         snapshot.getId());
   }
 
-  private Optional<Result> createGetResult(Snapshot.Key key, List<String> projections)
+  private Optional<Result> createGetResult(Snapshot.Key key, Get get, List<String> projections)
       throws CrudException {
     TableMetadata metadata = getTableMetadata(key.getNamespace(), key.getTable());
     return snapshot
-        .get(key)
+        .mergeResult(key, snapshot.get(get), get.getConjunctions())
```

Comment:
For get operations, the result and the write set state are merged while considering conjunctions.